mshv-bindings/src/unmarshal.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright © 2020, Microsoft Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+//
+use crate::bindings::*;
+
+// hv_message implementation for unmarshaling payload
+impl hv_message {
+    #[inline]
+    pub fn to_cpuid_info(&self) -> hv_x64_cpuid_intercept_message {
+        assert!(self.header.message_type == hv_message_type_HVMSG_X64_CPUID_INTERCEPT);
```

Comment:
Initially, we returned a result but based on @liuw suggestion we have used assert here because the caller needs to call exactly this function that that the exact type. there is no error handling here. live or die situation. :D

Reply:
I believe this should be left to the caller to decide if things should panic or not. I'm afraid unconditional asserts are also going to be frowned upon when submitting upstream.
I don't think this should be a blocker for merging this initial PR, but I'd appreciate if we could open an issue for that.

Reply:
@sameo Yes. I do agree. Lets @liuw comment on this as well.

Reply:
I don't feel strongly about this. Please change the code as you see fit.

Reply:
Issue: https://github.com/cloud-hypervisor/mshv/issues/4